config/nat.py

```diff
@@ -0,0 +1,1104 @@
+        if interface_name == interface:
+            tableFound = True
+    
+    if tableFound == False:
+        ctx.fail("Interface table is not present. Please configure ip-address on {} and apply the nat zone !!".format(interface_name)) 
+
+    config_db.mod_entry(interface_table_type, interface_name, {"nat_zone": nat_zone})
+
+#
+# 'nat remove interface' command ('config nat remove interface <interface_name>')
+#
+@remove.command('interface')
+@click.pass_context
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+def remove_interface(ctx, interface_name):
+    """Remove interface related NAT configuration"""
+    config_db = ConfigDBConnector()
+    config_db.connect()
+
+    if nat_interface_name_is_valid(interface_name) is False:
+        ctx.fail("Interface name is invalid. Please enter a  valid interface name!!")
+
+    if interface_name.startswith("Ethernet"):
+        interface_table_type = "INTERFACE"
+    elif interface_name.startswith("PortChannel"):
+        interface_table_type = "PORTCHANNEL_INTERFACE"
+    elif interface_name.startswith("Vlan"):
+        interface_table_type = "VLAN_INTERFACE"
+    elif interface_name.startswith("Loopback"):
+        interface_table_type = "LOOPBACK_INTERFACE"
+
+    interface_table_dict = config_db.get_table(interface_table_type)
+
+    if not interface_table_dict:
+        ctx.fail("Interface table is not present. Ignoring the nat zone configuartion")
+
+    for interface in interface_table_dict.keys():
+        if interface_name == interface:
+            tableFound = True
+
+    if tableFound == False:
+        ctx.fail("Interface table is not present. Ignoring the nat zone configuration")
+
+    config_db.mod_entry(interface_table_type, interface_name, {"nat_zone": "0"})
+
+#
+# 'nat remove interfaces' command ('config nat remove interfaces')
+#
+@remove.command('interfaces')
+@click.pass_context
+def remove_interfaces(ctx):
+    """Remove all interface related NAT configuration"""
+    config_db = ConfigDBConnector()
+    config_db.connect()
+
+    tables = ['INTERFACE', 'PORTCHANNEL_INTERFACE', 'VLAN_INTERFACE', 'LOOPBACK_INTERFACE']
+    nat_config = {"nat_zone": "0"}
+
+    for table_name in tables:
+        table_dict = config_db.get_table(table_name)
+        if table_dict:
+            for table_key_name in table_dict.keys():
+                if isinstance(table_key_name, unicode) is False:
+                    continue
+
+                config_db.set_entry(table_name, table_key_name, nat_config)
+
+#
+# 'nat feature' group ('config nat feature ')
+#
+@nat.group('feature')
+def feature():
+    """Enable or Disable the NAT feature"""
+    pass
+
+#
+# 'nat feature enable' command ('config nat feature enable>')
+#
+@feature.command('enable')
+@click.pass_context
+def enable(ctx):
+    """Enbale the NAT feature """
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    config_db.mod_entry("NAT_GLOBAL", "Values", {"admin_mode": "enabled"})
```

Comment:
When NAT is enabled but no timeout is configured, what will be the default timeout value ?

Reply:
For NAT entries, the default timeout is 600 seconds.
For NAPT entries, the default tcp timeout is 86400 seconds and udp timeout is 300 seconds.
When NAT is enabled, default timeouts are set in NatMgr code if timeouts are not configured before NAT enabled.